### ISSUE TITLE
getGrid.do returns error response as HTML

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/LAS#108]
**Reported by jonathan.s.callahan on 31 Jul 2007 17:02 UTC**
Working with the following dataset

Datasets > by Dataset Name > COADS monthly time series (1854-1993)   

When you ask for a comparison plot the SlideSorter begins initializing by requesting grid information from the following URL:

"getGrid.do?dsid=coads_time_series_1854_1993&varid=air"

The response from this request is not a JSON response as expected, but an HTML response that the javascript code in the client does not know how to handle (copied below)

It looks like the error has to do with passing in an invalid date string

```
Invalid format: "1854-1-15" is too short
```

but there is still the problem of getGrid.do returning an HTML error report instead of a JSON one.

```
<html><head><title>Apache Tomcat/5.5.17 - Error report</title><style><!--H1 {font-family:Tahoma,Arial

,sans-serif;color:white;background-color:#525D76;font-size:22px;} H2 {font-family:Tahoma,Arial,sans-serif

;color:white;background-color:#525D76;font-size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color

:white;background-color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-serif;color:black

;background-color:white;} B {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76

;} P {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;}A {color : black

;}A.name {color : black;}HR {color : #525D76;}--></style> </head><body><h1>HTTP Status 500 - </h1><HR

 size="1" noshade="noshade"><p><b>type</b> Exception report</p><p><b>message</b> <u></u></p><p><b>description

</b> <u>The server encountered an internal error () that prevented it from fulfilling this request.<

/u></p><p><b>exception</b> <pre>javax.servlet.ServletException: Invalid format: &quot;1854-1-15&quot

; is too short

	org.apache.struts.action.RequestProcessor.processException(RequestProcessor.java:523)

	org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:421)

	org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:224)

	org.apache.struts.action.ActionServlet.process(ActionServlet.java:1194)

	org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:414)

	javax.servlet.http.HttpServlet.service(HttpServlet.java:689)

	javax.servlet.http.HttpServlet.service(HttpServlet.java:802)

</pre></p><p><b>root cause</b> <pre>java.lang.IllegalArgumentException: Invalid format: &quot;1854-1-15

&quot; is too short

	org.joda.time.format.DateTimeFormatter.parseDateTime(DateTimeFormatter.java:683)

	gov.noaa.pmel.tmap.las.jdom.LASConfig.addTimeAxisAttributes(LASConfig.java:1098)

	gov.noaa.pmel.tmap.las.jdom.LASConfig.getGrid(LASConfig.java:1059)

	gov.noaa.pmel.tmap.las.ui.getGrid.execute(getGrid.java:78)

	org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:419)

	org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:224)

	org.apache.struts.action.ActionServlet.process(ActionServlet.java:1194)

	org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:414)

	javax.servlet.http.HttpServlet.service(HttpServlet.java:689)

	javax.servlet.http.HttpServlet.service(HttpServlet.java:802)

</pre></p><p><b>note</b> <u>The full stack trace of the root cause is available in the Apache Tomcat

/5.5.17 logs.</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/5.5.17</h3></body></html>```

Migrated-From: http://dunkel.pmel.noaa.gov/trac/las/ticket/104

**Comment by karlmsmith** [comment]
**Comment by jonathan.s.callahan on 31 Jul 2007 17:29 UTC**
I fixed the LASConfig.java typo which caused LAS to not recognize "1854-01-15" (r1583), but there is still the issue of returning HTML error responses to client code that expects JSON responses.

**Comment by karlmsmith** [comment]
**Comment by @noaaroland on 1 Aug 2007 16:43 UTC**
There has been no error response JSON object defined for the Ajax services.  A couple of weeks a go I asked Jeremy about this and he said he was happy with an empty object.

I can work on the services do a better job of exception handling so that the service does not response with the tomcat 505 error page, but instead response with an empty JSON object.

If we want some sort of formalization of a error object for these services we need to define it.  If it's as simple as reusing the XML error response from the product server I can do that.

Jeremy is in the driver's seat for this issue since it is his code that must deal with 99% of these messages.

**Comment by karlmsmith** [comment]
**Comment by jonathan.s.callahan on 1 Aug 2007 17:01 UTC**
'''I vote for reusing the existing error response!'''

**Comment by karlmsmith** [comment]
**Comment by @noaaroland on 24 Aug 2007 18:40 UTC**
Resolved by fixes for #120.
